ledger/accountdb_test.go

```diff
@@ -3761,3 +3761,103 @@ func TestUnfinishedCatchpointsTable(t *testing.T) {
 	}
 	require.Equal(t, expected, ret)
 }
+
+func TestRemoveOfflineStateProofID(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	dbs, _ := dbOpenTest(t, true)
+	setDbLogging(t, dbs)
+	defer dbs.Close()
+
+	tx, err := dbs.Wdb.Handle.Begin()
+	require.NoError(t, err)
+	defer tx.Rollback()
+
+	accts := ledgertesting.RandomAccounts(20, true)
+	for addr, acct := range accts {
+		rand.Read(acct.StateProofID[:])
+		accts[addr] = acct
+	}
+
+	// this is the same seq as accountsInitTest makes but it stops
+	// before the online accounts table creation to generate a trie and commit it
+	_, err = accountsInit(tx, accts, config.Consensus[protocol.ConsensusCurrentVersion])
+	require.NoError(t, err)
+
+	err = accountsAddNormalizedBalance(tx, config.Consensus[protocol.ConsensusCurrentVersion])
+	require.NoError(t, err)
+
+	err = accountsCreateResourceTable(context.Background(), tx)
+	require.NoError(t, err)
+
+	err = performResourceTableMigration(context.Background(), tx, nil)
+	require.NoError(t, err)
+
+	// create account hashes
+	computeRootHash := func() (crypto.Digest, error) {
+		rows, err := tx.Query("SELECT address, data FROM accountbase")
+		require.NoError(t, err)
+		defer rows.Close()
+
+		mc, err := MakeMerkleCommitter(tx, false)
+		require.NoError(t, err)
+		trie, err := merkletrie.MakeTrie(mc, TrieMemoryConfig)
+		require.NoError(t, err)
+
+		var addr basics.Address
+		for rows.Next() {
+			var addrbuf []byte
+			var encodedAcctData []byte
+			err = rows.Scan(&addrbuf, &encodedAcctData)
+			require.NoError(t, err)
+			copy(addr[:], addrbuf)
+			var ba baseAccountData
+			err = protocol.Decode(encodedAcctData, &ba)
+			require.NoError(t, err)
+			addHash := accountHashBuilderV6(addr, &ba, encodedAcctData)
+			added, err := trie.Add(addHash)
+			require.NoError(t, err)
+			require.True(t, added)
+		}
+		_, err = trie.Evict(true)
+		require.NoError(t, err)
+		return trie.RootHash()
+	}
+	oldRoot, err := computeRootHash()
+	require.NoError(t, err)
+	require.NotEmpty(t, oldRoot)
+
+	err = accountsCreateOnlineAccountsTable(context.Background(), tx)
+	require.NoError(t, err)
+	err = performOnlineAccountsTableMigration(context.Background(), tx, nil, nil)
+	require.NoError(t, err)
+
+	// get the new hash and ensure it does not match to the old one (data migrated)
+	mc, err := MakeMerkleCommitter(tx, false)
+	require.NoError(t, err)
+	trie, err := merkletrie.MakeTrie(mc, TrieMemoryConfig)
+	require.NoError(t, err)
+
+	newRoot, err := trie.RootHash()
+	require.NoError(t, err)
+	require.NotEmpty(t, newRoot)
+
+	require.NotEqual(t, oldRoot, newRoot)
```

Comment:
I tried to create an `expectedRoot` to compare with `newRoot` here but it didn't work. So I'm not sure if I did something wrong or something else is going on
https://github.com/nicholasguoalgorand/go-algorand/compare/nguo/catchpoint-stateproof-fix...cce:go-algorand:catchpoint-test-expected?w=1

Reply:
fixed thanks @algorandskiy